core/src/Module/Memory/DataSource/InMemoryDataSource.php

```diff
@@ -37,23 +40,14 @@ public function __construct(protected readonly string $genericClass) {
    * @throws QueryNotSupportedException
    */
   public function get(Query $query): mixed {
```

Comment:
I don't agree about removing the KeyQuery, it has it's own meaning and context

Reply:
Fixed.